lib/ansible/module_utils/lxd.py

```diff
@@ -93,7 +93,11 @@ def do(self, method, url, body_json=None, ok_error_codes=None, timeout=None):
         if resp_json['type'] == 'async':
             url = '{0}/wait'.format(resp_json['operation'])
             resp_json = self._send_request('GET', url)
-            if resp_json['metadata']['status'] != 'Success':
+            if 'status' in resp_json.get('metadata'):
+                status = resp_json['metadata']['status']
+            else:
+                status = resp_json.get('status', 'Failure')
```

Comment:
an empty status in metadata would still prevail, the following should 'safely' cover all cases

`status = resp_json.get('metadata', {}).get('status') or resp_json.get('status', 'Failure')`

Reply:
This part doesn't work ```resp_json.get('metadata', {}).get('status')``` and gives the below when running lxd_container in a playbook:

```
  File "/tmp/ansible_lxd_container_payload_elxsj3k0/ansible_lxd_container_payload.zip/ansible/module_utils/lxd.py", line 100, in do
AttributeError: 'str' object has no attribute 'get'
```